core/src/main/java/jenkins/slaves/JnlpSlaveAgentProtocol3.java

```diff
@@ -39,8 +39,7 @@
 
     @Override
     public String getName() {
-        if (ENABLED)    return "JNLP3-connect";
-        else            return "JNLP3-disabled";
+        return ENABLED ? "JNLP3-connect" : null;
```

Comment:
Should have been in a separate PR, but looks correct considered in isolation, according to super Javadoc.